components/camel-bindy/src/main/java/org/apache/camel/dataformat/bindy/BindyAbstractFactory.java

```diff
@@ -133,7 +133,7 @@ public abstract void bind(CamelContext camelContext, List<String> data, Map<Stri
     /**
      * Link objects together
      */
-    public void link(Map<String, Object> model) {
+    public synchronized void link(Map<String, Object> model) {
```

Comment:
If we use a second `BindyCsvDataFormat` in our route on the same model but with different parameters will it be good enough to fix the problem? Generally speaking, I'm wondering if it should not be fixed at `ReflectionHelper` level by simply not restoring the flag. WDYT?

Reply:
It was my first idea to fix it at ReflectionHelper level but I wanted my changes to impact as less as possible on other components, but yeah the scenario you described, even uncommon, can be a problem. I will change the commit.